Fix XCODE MTC Error

chris-bidwrangler opened the pull request:
## Description
Prior to fix xcode was bonking when running UITests 

Error received in xcode prior to fix: `[UIApplication applicationIconBadgeNumber] must be used from main thread
`
Followed documentation here for fix - https://developer.apple.com/documentation/code_diagnostics/main_thread_checker

## Related Issue
Seems to be similar to this issue - [https://github.com/phonegap/phonegap-plugin-push/issues/2901](url)

## Motivation and Context
Main Thread Checker is enabled by default in Xcode. If we manually disable the MTC the UITests work great, however since we are rebuilding the scheme when we rebuild the .ipa file this won't work for our automation purposes.

chris-bidwrangler committed:
Fix XCODE MTC Error

Error received prior to fix: [UIApplication applicationIconBadgeNumber] must be used from main thread

Followed documentation here for fix - https://developer.apple.com/documentation/code_diagnostics/main_thread_checker

aharbick commented on src/ios/PushPlugin.m:
```diff
@@ -225,7 +225,9 @@ - (void)init:(CDVInvokedUrlCommand*)command;
                 NSLog(@"PushPlugin.register: setting badge to true");
                 clearBadge = YES;
-                [[UIApplication sharedApplication] setApplicationIconBadgeNumber:0];
+                dispatch_sync(dispatch_get_main_queue(), ^{
```
This is a simpler change than I thought...  one question,  the link you sent in email seemed async.  Should it be ```dispatch_async```?

chris-bidwrangler replied:
Yes it should be. Thank you for catching that!

aharbick reviewed (approved):
Seems reasonably safe...  I’d definitely be sure to test push notification and badge functionality with this build though.

chris-bidwrangler committed:
Fix typo

changed dispatch_sync(dispatch_get_main_queue(), ^{ to be dispatch_async(dispatch_get_main_queue(), ^{
